packages/kbn-config-schema/src/internals/index.ts

```diff
@@ -46,19 +46,23 @@ const anyCustomRule: Rules = {
       .required(),
   },
   validate(params, value, state, options) {
-    let validationResultMessage;
+    let validationResult: CustomValidatorResult;
     try {
-      validationResultMessage = params.validator(value);
+      validationResult = params.validator(value);
     } catch (e) {
-      validationResultMessage = e.message || e;
+      validationResult = e.message || e;
     }
 
-    if (typeof validationResultMessage === 'string') {
-      return this.createError(
-        'any.custom',
-        { value, message: validationResultMessage },
-        state,
-        options
+    if (typeof validationResult === 'string') {
+      return this.createError('any.custom', { value, message: validationResult }, state, options);
+    } else if (typeof validationResult === 'object') {
```

Comment:
I'm allowing custom validators to return a more detailed object now in order to retain some of the information provided by some of the richer Joi internal validators.